clippy.toml

```diff
@@ -1 +1,3 @@
-type-complexity-threshold = 500
+type-complexity-threshold = 750
+cognitive-complexity-threshold = 50
+too-many-arguments-threshold = 15
```

Comment:
Can you explain how you came up with these values? I don't think having functions with 15 arguments is a very good idea, and the cognitive complexity I think should be lowered if anything, since the default is already really high.

If type complexity causes issues with specs types, I think we should either mark those types with `[allow(clippy::type_complexity)]` or disable the lint entirely. Preferably the former since we would still like non-specs types that are very complex to cause warnings.

Reply:
These are the minimum values to get amethyst to pass the lints. If we want to lower them, it takes A. Significant refactoring, or B. Nonsensical refactoring or allowances such as with specs. Arguing about the exact values in an initial pass PR where the goal was to just get amethyst passing with future work to meet better lints (pedantic) is ..... Pedantic

this is not a pass to make significant changes everywhere to pass lints. I've made simple corrections where possible, and otherwise allowed lints where not. Because we now have flags for allowances, they can be fixed at a future time. 

If future work on the lints is not acceptable, then it's best to remove all this and close the PR.